src/client/features/interactions/index.js

```diff
@@ -56,171 +56,66 @@ class Interactions extends React.Component {
+        ids : metaData.ids,
+        loaded:_.assign(this.state.loaded,{network:true, ids:true})
       });
-      const initialLayoutOpts = state.layoutConfig.defaultLayout.options;
-      const layout = cy.layout(initialLayoutOpts);
-      layout.run();
-    });
-  }
-
-  edgeType(type){
-    switch(type){
-      case 'in-complex-with':
-      case 'interacts-with':
-        return 'Binding';
-      case 'controls-phosphorylation-of':
-        return 'Phosphorylation';
-      case 'controls-expression-of':
-        return 'Expression';
-      default:
-        return '';
-    }
-  }
-
-  interactionMetadata(mediatorIds,pubmedIds){
-    let metadata = [['List',[]],['Detailed Views',[]]];//Format expected by format-content
-    mediatorIds.split(';').forEach( link => {
-      const id=link.split('/')[4];
-      metadata[1][1].push(link.includes('reactome') ? ['Reactome',id]:['Pathway Commons',id]);
-    });
-    if(pubmedIds){
-     pubmedIds.split(';').forEach(id=>metadata[0][1].push(['PubMed',id]));
-    }
-   return metadata;
-}
-
-  addInteraction(nodes,edge,sources,network,nodeMap,nodeMetadata){
-    const interaction= this.edgeType(edge);
-    nodes.forEach((node)=>{
-      if(!nodeMap.has(node)){
-        const metadata=nodeMetadata.get(node);
-        nodeMap.set(node,true);
-        const links=_.uniqWith(_.flatten(metadata.slice(-2).map(entry => entry.split(';').map(entry=>entry.split(':')))),_.isEqual).filter(entry=>entry[0]!='intact');
-        network.nodes.push({data:{class: "ball",id: node,label: node, queried: this.state.ids.indexOf(node)!=-1 ,
-        parsedMetadata:[['Type','bp:'+metadata[0].split(' ')[0].replace(/Reference/g,'').replace(/;/g,',')],['Database IDs', links]]}});
-      }
-    });
-
-    network.edges.push({data: {
-      id: nodes[0]+'\t'+edge+'\t'+nodes[1] ,
-      label: nodes[0]+' '+edge.replace(/-/g,' ')+' '+nodes[1] ,
-      source: nodes[0],
-      target: nodes[1],
-      class: interaction,
-      parsedMetadata:sources
-    },classes:interaction});
-  }
-
-  parse(data){
-    let network = {
-      edges:[],
-      nodes:[],
+  });
+
+   this.state.cy.on('trim', () => {
+     const state = this.state;
+     const ids = state.ids;
+     if(ids.length === sources.length){
+       const cy = state.cy;
+       const mainNode = cy.nodes(node=> ids.indexOf(node.data().id) != -1);
+       const nodesToKeep = mainNode.merge(mainNode.connectedEdges().connectedNodes());
+       cy.remove(cy.nodes().difference(nodesToKeep));
+     }
+   });
+
+   this.state.cy.one('layoutstop',()=>{
+     const state = this.state;
+     const cy = this.state.cy;
+     const categories = state.categories;
+     const filters=state.filters;
+     _.forEach(filters,(value,type)=>{
+       const edges = cy.edges().filter(`.${type}`);
+       const nodes = edges.connectedNodes();
+       edges.length?
+       categories.set(type,{edges:edges,nodes:nodes}):
+       (categories.delete(type),delete filters[type]);
```

Comment:
What does `(categories.delete(type),delete filters[type]);` accomplish?

Reply:
It is deleting unused filter. We have three filters (binding, phosphorylation and expression). If none of the nodes falls in one filter, it will be removed. I have not modified the actual "front end" code yet. I guess the function should be working.